test/deprecates.jl

```diff
@@ -0,0 +1,201 @@
+using StatsBase
+using Base.Test
+using Compat
+import Compat: view
+
+@testset "StatsBase.Deprecates" begin
+
+@testset "Deprecates WeightVec and weights" begin
```

Comment:
Unfortunately, doing this will print lots of deprecation warnings, which will make the output hard to read, and deprecations from Base that need to be fixed won't be easy to spot. As long as you've tested it once locally, it's OK.